File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        if (nums.length == 1) return 1;
        int start_pos = 0;
        int index = 0;
        while (index < nums.length) {
            int current = nums[index];
            // find duplicates for current;
            while (index < nums.length && nums[index] == current) {
                index++;
            }
            nums[start_pos++] = current;
        }
        return start_pos;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        index = -1
        prev = None
        for i in range(len(nums)):
            if index == -1 or prev != nums[i]:
                index += 1
                prev = nums[i]
                nums[index] = nums[i]
        return index + 1